Update Android FHIR SDK library versions

rkareko opened the pull request:
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes [link to issue]

Fixed issues contained in this release
[Move the add button for repeated groups from top to bottom](https://github.com/google/android-fhir/issues/2844)
[Translation is not coming in validation dialog after submission of the Questionnaire](https://github.com/google/android-fhir/issues/2886)
[InitialExpression with result of type Coding should ignore display](https://github.com/google/android-fhir/issues/2880)
[Migrate Review Recyclerview to Compose](https://github.com/google/android-fhir/issues/2852)
[Migrate EditTextViewHolderFactory to compose](https://github.com/google/android-fhir/issues/2855)

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._


**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

rkareko committed:
Bump up SDC library version

rkareko committed:
Update library versions

rkareko committed:
Fix compile time errors

rkareko committed:
Merge branch 'main' into update-sdc-lib-snapshot

rkareko committed:
Merge branch 'main' into update-sdc-lib-snapshot

FikriMilano reviewed (changes requested):
the textbox's question text and spacing is a bit broken when using this snapshot.
we'll need to investigate.

<img width="200" alt="Image" src="https://github.com/user-attachments/assets/f39cd3d7-9a87-4e99-8caa-e0f261b34687" />

<img width="200" alt="Image" src="https://github.com/user-attachments/assets/e99e8248-f236-4632-ba32-e9597d6d7c32" />

LZRS committed:
Merge remote-tracking branch 'origin/main' into update-sdc-lib-snapshot

LZRS committed:
Update Questionnaire population with sdc changes on immutability

Fixes https://github.com/opensrp/fhircore/issues/3766#issuecomment-3616145609
where Questionnaire is no longer mutated during population

after https://github.com/google/android-fhir/pull/2887 merge

ajhmohammed commented:
@Rkareko @LZRS Any updates on this PR?

LZRS committed:
Update to latest sdc version 1.3.0-preview10-SNAPSHOT

LZRS committed:
Merge remote-tracking branch 'origin/main' into update-sdc-lib-snapshot
